curriculum/path-data.json

```diff
@@ -0,0 +1,2070 @@
+{
+  "subPathToPageId": {
+    "2022/responsive-web-design/build-a-personal-portfolio-webpage-project/build-a-personal-portfolio-webpage": "HzgW",
+    "2022/responsive-web-design/build-a-product-landing-page-project/build-a-product-landing-page": "Bdc7",
+    "2022/responsive-web-design/build-a-survey-form-project/build-a-survey-form": "vKMf",
+    "2022/responsive-web-design/build-a-technical-documentation-page-project/build-a-technical-documentation-page": "8z6c",
+    "2022/responsive-web-design/build-a-tribute-page-project/build-a-tribute-page": "spDx",
+    "2022/responsive-web-design/learn-accessibility-by-building-a-quiz/step-1": "MVPV",
+    "2022/responsive-web-design/learn-accessibility-by-building-a-quiz/step-10": "xgV7",
+    "2022/responsive-web-design/learn-accessibility-by-building-a-quiz/step-11": "LDf9",
+    "2022/responsive-web-design/learn-accessibility-by-building-a-quiz/step-12": "ZFGW",
```

Comment:
What happens here when we insert a new step and everything shifts?

Reply:
I think we'd need to shift them with it. We _could_ just insert a new page and keep the old ids, but that feels more confusing to me.

Reply:
To clarify: if we insert a new step 10 then that would get a random id. The old step 10 would keep xgV7, but be called step 11 and so on.

In other words, I think the content is most important, not the title. It should cause less confusion on the forums to do it this way.

Reply:
I feel like there's a lot of room for error there.

Reply:
There is, but I'm not sure how to implement the redirection we want without doing something like this.

That said, we're going to break other superblock back links at some point, so we can park this for a while and try to come up with something smarter.

Reply:
What if instead of mapping a path to a fCCID (it's UUID but ours) we map the challenge mongoID?

Reply:
I'm not sure how that would work. We'd need to be able to map superblocks (which don't have mongo ids) and duplicate challenges (which share mongo ids). What did you have in mind?

One worry I have is that this would mix concerns. Mongo ids currently just identify user progress. If we map between them and fCCIDs they would take on an additional role.

Reply:
> We'd need to be able to map superblocks

Why not a separate map for those?

>  Mongo ids currently just identify user progress

They also serve as a unique identifier.

My thought process is that at some point, we're going to make `/learn/naomi` work as a slug to point to whatever page it should. If we map superblocks and map challenges (based on mongoID), that UUID could resolve to whatever it needs to.

The challenge path might change, such as when we reorder steps, but the UUID would be tied to the MongoID for the challenge which is immutable - we wouldn't have to update anything when we insert a step.

Reply:
> Why not a separate map for those?

Could do, but then we'd have mongo ids that map to fCCIds and mongo ids that map to fCCIds and challenge steps. It's a little messy.

> They also serve as a unique identifier.

Right, but each mongo id might need to map to more than one fCCId, which they can't.

Even if we didn't have this constraint right now, we still should keep the ids separate because it reduces our flexibility.